packages/fx-core/src/plugins/resource/bot/dotnet/errors.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getLocalizedString, getDefaultString } from "../../../../common/localizeUtils";
+import { ErrorType, PluginError } from "../errors";
+
+const tips = {
+  checkLog: getLocalizedString("plugins.dotnet.checkLog"),
+  checkFsPermissions: getLocalizedString("plugins.dotnet.checkFsPermissions"),
+};
+
+export class DotnetPluginError extends PluginError {
+  public innerError?: Error;
+
+  constructor(
+    errorType: ErrorType,
+    code: string,
+    messages: [string, string],
+    suggestions: string[],
+    helpLink?: string,
+    innerError?: Error
+  ) {
+    super(errorType, code, messages, suggestions, helpLink);
+    this.innerError = innerError;
+  }
+}
+
+export class FileIOError extends DotnetPluginError {
```

Comment:
Since it is still Bot plugin but dotnet implementation, is that possible to leverage the existing PluginError or FileIOError from bot level instead of creating new class?